### ISSUE TITLE
Add CI builds for GLX and OSMesa

### ISSUE THREAD

**Issue by kkaefer** [mapbox/mapbox-gl-native#10566]
We're currently only testing EGL. We have code that supports GLX and OSMesa as well, but there are no automated tests for them, which means we're more likely to add a regression.

**Comment by kkaefer** [comment]
Alternatively, we can also remove GLX and OSMesa support. @tmpsantos @brunoabinader @bsudekum any reason to retain it?

**Comment by brunoabinader** [comment]
We decided to test EGL only as part of our plans to reduce CI jobs, back when we were using Travis CI. 

OSMesa was added as an off-screen rendering alternative for running CI jobs in machines w/o GPU acceleration support - its biggest advantage in comparison to e.g. EGL is that it doesn't require a running window system. It might still be relevant to keep this around to address scenarios where neither GPU nor window systems are available.

GLX is still the current default GL provider in our build system - even though all our Linux-based CI jobs use EGL nowadays. Like the OSMesa case, we might want to keep it around in case we need to support old platforms that don't provide EGL support.

I don't have a strong argument against removing both, but I'd appreciate if we could find a way to keep them alive somehow.
